sophia/src/term/blank_node.rs

```diff
@@ -0,0 +1,287 @@
+}
+
+/// Internal representation of a blank node identifier.
+///
+/// May be encountered when pattern-matching on [`Term`](enum.Term.html)s
+/// of the [`BNode`](enum.Term.html#variant.BNode) variant.
+/// For that purpose, note that `BlankNode`
+///  - derefs implicitly to its internal type `T`;
+///  - can be directly compared to a `&str` with the `==` operator.
+///
+/// Example :
+/// ```
+/// use sophia::term::*;
+///
+/// fn is_foobar(t: BoxTerm) -> bool {
+///     match t {
+///         BNode(bn) =>
+///             bn.starts_with("foo") || bn == "bar",
+///         _ =>
+///             false,
+///     }
+/// }
+/// ```
+///
+/// See [module documentation](index.html)
+/// for more detail.
+#[derive(Clone, Copy, Debug, Eq)]
+pub struct BlankNode<TD: TermData>(TD);
+
+impl<TD> BlankNode<TD>
+where
+    TD: TermData,
+{
+    /// Return a new blank node with the given identifier.
+    ///
+    /// May fail if `id` is not a valid identifier according to
+    /// [`BLANK_NODE_LABEL`](struct.BLANK_NODE_LABEL.html). This means that it
+    /// must not include the typical leading `_:`.
+    pub fn new<U>(id: U) -> Result<Self>
+    where
+        U: AsRef<str>,
+        TD: From<U>,
+    {
+        if BLANK_NODE_LABEL.is_match(id.as_ref()) {
+            Ok(BlankNode(id.into()))
+        } else {
+            Err(TermError::InvalidBlankNodeId(id.as_ref().to_string()))
+        }
+    }
+
+    /// Return a new blank node with the given identifier.
+    ///
+    /// # Safety
+    ///
+    /// This function requires that `id` is a valid blank node identifier.
+    pub unsafe fn new_unchecked<U>(id: U) -> Self
+    where
+        TD: From<U>,
+    {
+        BlankNode(id.into())
+    }
+
+    /// Copy self while transforming the inner `TermData` with the given
+    /// factory.
+    pub fn copy_with<'a, U, F>(&'a self, mut factory: F) -> BlankNode<U>
+    where
+        U: TermData,
+        F: FnMut(&'a str) -> U,
+    {
+        BlankNode(factory(self.as_ref()))
+    }
+
+    /// Writes the blank node to the `fmt::Write` using the N3 syntax.
+    pub fn write_fmt<W>(&self, w: &mut W) -> fmt::Result
+    where
+        W: fmt::Write,
+    {
+        w.write_str("_:")?;
+        w.write_str(self.as_ref())
+    }
+
+    /// Writes the blank node to the `io::Write` using the N3 syntax.
+    pub fn write_io<W>(&self, w: &mut W) -> io::Result<()>
+    where
+        W: io::Write,
+    {
+        w.write_all(b"_:")?;
+        w.write_all(self.as_ref().as_bytes())
+    }
+
+    /// Return a copy of this blank nodes's underlying identifier.
+    pub fn value(&self) -> String {
+        self.as_ref().to_owned()
+    }
+
+    /// Whether this blank node identifier is compatible with the N3 family
+    /// of syntaxes (N-Triples, Turtle...).
+    ///
+    /// This is already checked by the standard constructor
+    /// [`new()`](struct.BlankNode.html#mehod.new).
+    pub fn is_n3(&self) -> bool {
+        BLANK_NODE_LABEL.is_match(self.as_ref())
+    }
```

Comment:
This method is moot now that we block non-N3 identifier in the constructor. Probably good to remove it completely.

Reply:
You're right I removed it.